web/app/scripts/filterbar/date-range-directive.js

```diff
@@ -2,54 +2,115 @@
+                        }
+                        if (dtRange.max) {
+                            var jsMax = moment(dtRange.max, moment.ISO_8601).toDate();
+                            scope.max = calendar.formatDate(
+                                scope.calendarOptions.dateFormat,
+                                calendar.fromJSDate(jsMax)
+                            );
+                            $('#dtMaxField').calendarsPicker('setDate', calendar.fromJSDate(jsMax));
+                        }
                         scope.isMinMaxValid();
                     }
                 });
-                scope.$watch('dtMin', function(newMin) {
-                    $('#dtMinField').datepicker('update', newMin);
-                    scope.onDtRangeChange();
-                });
-                scope.$watch('dtMax', function(newMax) {
-                    $('#dtMaxField').datepicker('update', newMax);
-                    scope.onDtRangeChange();
-                });
 
-                // On change of DT value
-                scope.onDtRangeChange = function() {
-                    dtRange.min = scope.dtMin;
-                    dtRange.max = scope.dtMax;
+                /**
+                 * Update the scope and dtRange date variables whenever a new date is selected
+                 * in the calendarPicker. This is fired by the calendar's onSelect action.
+                 * @param date {CDate} The newly selected date, as a CDate.
+                 * @param minOrMax {string} A string specifying 'min' or 'max'.
+                 */
+                function updateDate(date, minOrMax) {
+                    scope[minOrMax] = calendar.formatDate(scope.calendarOptions.dateFormat, date);
+                    var gregCDate = DateLocalization.convertToCalendar(date, 'gregorian', 'en');
+                    var gregISODate = gregCDate._calendar.toJSDate(gregCDate).toJSON();
+                    dtRange[minOrMax] = gregISODate;
                     scope.updateFilter();
-                };
+                }
+
+                /**
+                 * Sets up the calendar picker based on the current language.
+                 */
+                function configureDatePicker() {
+                    var dateConfig = DateLocalization.currentDateFormats();
+                    calendar = $.calendars.instance(dateConfig.calendar, dateConfig.language);
+                    $.calendarsPicker.setDefaults($.calendarsPicker.regionalOptions['']);
+                    scope.calendarOptions = angular.extend({
+                        calendar: calendar,
+                        dateFormat: dateConfig.formats.numeric,
+                        showAnim: '',
+                    }, $.calendarsPicker.regionalOptions[dateConfig.language]);
 
+                    // Make sure that the datepicker hides when the parent dropdown does
+                    elem.on('hide.bs.dropdown', function() {
```

Comment:
Seems like you just need to call `stopPropagation` on the datepicker element or its wrapper (if there is one). I'm not familiar with this API, but that should prevent the click bubbling up to the dropdown. The best way to do that would be to add a separate click handler here that just calls stopPropagation.

This fix seems ok as is, just a bit brittle because it depends on a particular dom ID. Could do a `$document.find('.datepicker-class')` using a class you add to all datepicker elements, rather than calling the same method on a bunch of id elements here.